crates/compiler/collections/src/soa.rs

```diff
@@ -184,4 +184,10 @@ impl<T, U> EitherIndex<T, U> {
             Err(Index::new(self.index ^ Self::MASK))
         }
     }
+
+    pub fn decrement_index(&mut self) {
+        if self.index > 0 {
+            self.index -= 1;
+        }
```

Comment:
This can be `self.index = self.index.saturating_sub(1)` - https://doc.rust-lang.org/std/primitive.usize.html#method.saturating_sub